test/Makefile

```diff
@@ -0,0 +1,64 @@
+LIBDIR=../lib
+GTEST_MODULE=$(LIBDIR)/googletest
+GTEST_DIR=$(LIBDIR)/googletest/googletest
+GTEST_SOURCEDIR=$(GTEST_DIR)/src
+GTEST_HEADERS=$(GTEST_DIR)/include/gtest/*.h \
+              $(GTEST_DIR)/include/gtest/internal/*.h
+GTEST_SRCS_=$(GTEST_SOURCEDIR)/*.cc $(GTEST_SOURCEDIR)/*.h $(GTEST_HEADERS)
+
+CPP_NETLIB_DIR=$(LIBDIR)/cpp-netlib
+YAML_CPP_DIR=$(LIBDIR)/yaml-cpp
+
+CPPFLAGS+=-isystem $(GTEST_DIR)/include
+CXXFLAGS=-std=c++11 -g -pthread
+
+# Where to find user code.
+SRC_DIR=../src
+SRCS=\
+     $(SRC_DIR)/format.cc
+OBJS=$(subst src/,test/,$(subst .cc,.o,$(SRCS)))
+
+TEST_DIR=.
+TESTS=$(wildcard $(TEST_DIR)/*.cc)
+TEST_OBJS=$(TESTS:%.cc=%.o)
+
+GTEST_LIBS=gtest_main.a
+
+all: build-googletest unittest
+
+clean:
+	rm -f unittest *.a *.o
+
+purge: clean
+	git submodule deinit -f --all
+
+init-submodules:
+	git submodule init && \
+	    git submodule update
+
+$(GTEST_SRCS_): init-submodules
+$(GTEST_DIR)/Makefile: $(GTEST_SRCS_)
+	$(CXX) $(CPPFLAGS) -I$(GTEST_DIR) $(CXXFLAGS) -c \
+	    $(GTEST_SOURCEDIR)/gtest-all.cc
+	$(CXX) $(CPPFLAGS) -I$(GTEST_DIR) $(CXXFLAGS) -c \
+	    $(GTEST_SOURCEDIR)/gtest_main.cc
+
+$(GTEST_LIBS): build-googletest
```

Comment:
Why have a phony target in the middle? This will re-archive every time. Just make `$(GTEST_LIBS)` depend on `$(GTEST_DIR)/Makefile`, `gtest_main.a` and the object files. Or make the target an actual file.

Reply:
I made the target an actual file - now the only phony targets are clean, purge and all

Reply:
If that file is never created (and cleaned up), it might as well be a phony target... I was proposing this:
```Makefile
$(GTEST_LIBS): $(GTEST_DIR)/Makefile
	$(AR) $(ARFLAGS) gtest_main.a gtest-all.o gtest_main.o
```